refapp/SKWebApi/Program.cs

```diff
@@ -1,50 +1,92 @@
+            .AddMicrosoftIdentityWebApi(builder.Configuration.GetSection("AzureAd"));
+        builder.Services.AddAuthorization();*/
+	
+        services.AddControllers();
+        // Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
+        services.AddEndpointsApiExplorer();
+        services.AddSwaggerGen();
 
-//var scopeRequiredByApi = app.Configuration["AzureAd:Scopes"] ?? "";
-//var summaries = new[]
-//{
-//    "Freezing", "Bracing", "Chilly", "Cool", "Mild", "Warm", "Balmy", "Hot", "Sweltering", "Scorching"
-//};
+        services.AddSingleton<IConfiguration>(configuration);
 
-//app.MapGet("/weatherforecast", (HttpContext httpContext) =>
-//{
-//    httpContext.VerifyUserHasAnyAcceptedScope(scopeRequiredByApi);
+        services.AddSingleton<ILogger>(s => s.GetRequiredService<ILogger<Kernel>>()); // To support ILogger (as opposed to generic ILogger<T>)
 
-//    return forecast;
-//})
-//.WithName("GetWeatherForecast")
-//.RequireAuthorization();
+        services.AddSemanticKernelServices(configuration);
+    }
 
-app.Run();
+    private static void AddSemanticKernelServices(this IServiceCollection services, ConfigurationManager configuration)
+    {
+        // Each REST call gets a fresh new SK instance
+        var kernelConfig = new KernelConfig();
+        AIServiceConfig completionConfig = configuration.GetRequiredSection("CompletionConfig").Get<AIServiceConfig>();
+        kernelConfig.AddCompletionBackend(completionConfig);
+        ISemanticTextMemory memory = NullMemory.Instance;
+        AIServiceConfig embeddingConfig = configuration.GetSection("EmbeddingConfig").Get<AIServiceConfig>();
+        if (embeddingConfig.IsValid())
+        {
+            // The same SK memory store is shared with all REST calls and users
+            IMemoryStore<float> memoryStore = new VolatileMemoryStore();
+            IEmbeddingGenerator<string, float> embeddingGenerator = embeddingConfig.ToTextEmbeddingsService(/* TODO: add logger - Might need to make SK classes more amenable to DI to do this... */);
+            kernelConfig.AddEmbeddingBackend(embeddingConfig);
+#pragma warning disable CA2000 // Dispose objects before losing scope - Used later through DI
+            memory = new SemanticTextMemory(memoryStore, embeddingGenerator);
+#pragma warning restore CA2000 // Dispose objects before losing scope
+        }
+        services.AddSingleton<ISkillCollection, SkillCollection>(); // Keep skill list empty?
+        services.AddSingleton<IPromptTemplateEngine, PromptTemplateEngine>();
+        services.AddSingleton(memory);
+        services.AddSingleton(kernelConfig);
+
+        services.AddScoped<Kernel>();
+
+        IDictionary<string, Type> mapOfDependenciesToTypes = new Dictionary<string, Type>();
+        string nativeSkillsDirectory = configuration.GetSection(Constants.NativeSkillsDirectory).Get<string>();
+        if (!string.IsNullOrWhiteSpace(nativeSkillsDirectory))
```

Comment:
From our offline conversation - let's start with a hardcoded skill example and save this plugin loading model for a future update.

Reply:
Done